src/tools/gcc.lua

```diff
@@ -236,8 +236,20 @@
 
 	gcc.libraryDirectories = {
 		architecture = {
-			x86 = "-L/usr/lib32",
-			x86_64 = "-L/usr/lib64",
+			x86 = function (cfg)
+				local r = {}
+				if cfg.system ~= premake.MACOSX then
+					table.insert (r, "-L/usr/lib32")
+				end
+				return r
+			end,
+			x86_64 = function (cfg)
+				local r = {}
+				if cfg.system ~= premake.MACOSX then
+					table.insert (r, "-L/usr/lib64")
+				end
+				return r
+			end,
```

Comment:
I guess it makes more sense to be a table, it allows modules to just override the function and assume a table. Either way, we need to figure out a way to not use a function every time an OS differs to the others. While it's a bit over the top, it would be nice if there was something like this:
```lua
architecture = {
  x86 = {
    default = { "-L/usr/lib32" },
    OSX = { }
  },
  x86_64 = {
    default = { "-L/usr/lib64" },
    OSX = { }
  }
}
```

This would allow us to quite easily support Clang on Windows (I'm assuming nmake does/will support Clang).